### PR TITLE
Update settings tab icons

### PR DESCRIPTION
**Pull request by dbkr**
...and add support for compound design token icons to TabbedView, changing all the other icons over while we're at it.

To go with https://github.com/matrix-org/matrix-react-sdk/pull/12841

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

File: src/components/structures/TabbedView.tsx [hunk 1]
```diff
@@ -98,7 +98,9 @@ function TabLabel<T extends string>({ tab, isActive, showToolip, onClick }: ITab
     });
 
     let tabIcon: JSX.Element | undefined;
-    if (tab.icon) {
+    if (tab.icon && typeof tab.icon === "object") {
+        tabIcon = tab.icon;
+    } else if (typeof tab.icon === "string") {
         tabIcon = <span className={`mx_TabbedView_maskedIcon ${tab.icon}`} />;
     }
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/structures/TabbedView.tsx:105]: `null` would also hit this edge, no?

[the remark this answers is not in the document]
Comment (dbkr) [reply src/components/structures/TabbedView.tsx]: `null` is of type `object` so it wouldn't match either case.

File: src/components/views/dialogs/UserSettingsDialog.tsx [hunk 2]
```diff
@@ -17,6 +17,18 @@ limitations under the License.
...
+import { Icon as VisibilityOnIcon } from "@vector-im/compound-design-tokens/icons/visibility-on.svg";
+import { Icon as NotificationsIcon } from "@vector-im/compound-design-tokens/icons/notifications.svg";
+import { Icon as PreferencesIcon } from "@vector-im/compound-design-tokens/icons/preferences.svg";
+import { Icon as KeyboardIcon } from "@vector-im/compound-design-tokens/icons/keyboard.svg";
+import { Icon as SidebarIcon } from "@vector-im/compound-design-tokens/icons/sidebar.svg";
+import { Icon as MicOnIcon } from "@vector-im/compound-design-tokens/icons/mic-on.svg";
+import { Icon as LockIcon } from "@vector-im/compound-design-tokens/icons/lock.svg";
+import { Icon as LabsIcon } from "@vector-im/compound-design-tokens/icons/labs.svg";
+import { Icon as BlockIcon } from "@vector-im/compound-design-tokens/icons/block.svg";
+import { Icon as HelpIcon } from "@vector-im/compound-design-tokens/icons/help.svg";
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/dialogs/UserSettingsDialog.tsx]: Indeed, we should probably eslint ban the source svgs

File: src/components/structures/TabbedView.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/structures/TabbedView.tsx]: Yes but it isn't truthy so `(tab.icon && typeof tab.icon === "object") === false`

**Review by t3chguy — APPROVED** [review]

**Pull request merged**